src/Adapters/Auth/huawei.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+const https = require('https');
+var Parse = require('parse/node').Parse;
+
+const TOKEN_ISSUER = "https://accounts.huawei.com";
+const HUAWEI_URL =   "https://oauth-login.cloud.huawei.com/oauth2/v3/tokeninfo";
+
+
+function getDataFromHuawei(id_token){
+  return new Promise((resolve, reject) => {
+    https.get(`${HUAWEI_URL}?id_token=${id_token}`, res => {
+      let data = '';
+      res.on('data', chunk => {
+        data += chunk.toString('utf8');
+      });
+      res.on('end', () => {
+        resolve(JSON.parse(data));
+      });
+    }).on('error', reject);
+  });
+}
+
+
+async function verifyIdToken({ id_token: token, id }) {
+  if (!token) {
+    throw new Parse.Error(Parse.Error.OBJECT_NOT_FOUND, `id token is invalid for this user.`);
```

Comment:
Please also capitalize this and all other error messages:
> ID token is invalid for this user.

"ID token" looks confusing, because there seems to be an ID and a token and these do not appear to be the same?

Reply:
Sorry, I misunderstood that you said me to change "id token" to "ID token"
Now I change all "ID token" to "id token"

Reply:
`ID token` was correct as you had it, but I wonder about the term "ID token`:

`{ id_token: token, id }`

There are 2 parameters, a `token`and an `id`. I suggest to also write only "Token is invalid for this user." Because the "id_token` seems to mean "a token for an ID".

Reply:
I completely understand
Thank you for your kind comment!!